source/rpt/rpt_cell_reconstruction.cc

```diff
@@ -0,0 +1,626 @@
+    {
+      myfile
+        << "unknown_positions_id volumes particle_positions_x particle_positions_y particle_positions_z cell_level";
+      sep = " ";
+    }
+  else // .csv is default
+    {
+      myfile
+        << "unknown_positions_id,volumes,particle_positions_x,particle_positions_y,particle_positions_z,cell_level";
+      sep = ",";
+    }
+
+  // Add status columns and distance if analyse positions is enable
+  if (parameters.reconstruction_param.analyse_positions)
+    myfile << sep << "status" << sep << "status_id" << sep << "distance"
+           << std::endl;
+  else
+    myfile << std::endl;
+
+  for (unsigned int i_particle = 0;
+       i_particle < reconstruction_positions.size();
+       i_particle++)
+    {
+      myfile << i_particle << sep << cells_volumes[i_particle] << sep
+             << reconstruction_positions[i_particle][0] << sep
+             << reconstruction_positions[i_particle][1] << sep
+             << reconstruction_positions[i_particle][2] << sep
+             << final_cell_level[i_particle];
+
+      if (parameters.reconstruction_param.analyse_positions)
+        {
+          // Calculate distance between known and found positions
+          double distance =
+            std::sqrt(std::pow(reconstruction_positions[i_particle][0] -
+                                 known_positions[i_particle][0],
+                               2) +
+                      std::pow(reconstruction_positions[i_particle][1] -
+                                 known_positions[i_particle][1],
+                               2) +
+                      std::pow(reconstruction_positions[i_particle][2] -
+                                 known_positions[i_particle][2],
+                               2));
+
+          // Associate status to a status id
+          unsigned int status_id;
+          if (cell_status[i_particle] == "right_cell")
+            status_id = 0;
+          else if (cell_status[i_particle] == "cost_function|right_cell")
+            status_id = 1;
+          else if (cell_status[i_particle] == "parent_cell|right_cell")
+            status_id = 2;
+          else if (cell_status[i_particle] ==
+                   "parent_cell|cost_function|right_cell")
+            status_id = 3;
+          else if (cell_status[i_particle] == "wrong_cell")
+            status_id = 4;
+          else if (cell_status[i_particle] == "cost_function|wrong_cell")
+            status_id = 5;
+          else if (cell_status[i_particle] == "parent_cell|wrong_cell")
+            status_id = 6;
+          else if (cell_status[i_particle] ==
+                   "parent_cell|cost_function|wrong_cell")
+            status_id = 7;
+
+
+          myfile << sep << cell_status[i_particle] << sep << status_id << sep
+                 << distance << std::endl;
+        }
+      else
+        myfile << std::endl;
+    }
+}
+
+template <int dim>
+void
+RPTCellReconstruction<dim>::assign_detector_positions()
```

Comment:
This is also present in the rpt.cc file no? It would be better to move it to a seperate file and define it as a function instead.

Reply:
Good idea!